common/src/main/java/com/wynntils/core/consumers/overlays/Overlay.java

```diff
@@ -63,6 +75,28 @@ protected Overlay(
         this.verticalAlignmentOverride.store(verticalAlignmentOverride);
     }
 
+    /**
+     * Whether the overlay should be hidden when Wynncraft hides gui.
+     */
+    protected boolean hideWhenNoGui() {
+        return true;
+    }
+
+    /**
+     * Whether the overlay should be rendered.
+     */
+    protected boolean isVisible() {
+        return true;
+    }
+
+    protected boolean isRendered() {
+        if (enabledTemplateCache == null || enabledTemplateCache.hasError())
+            return ((Models.WorldState.onWorld() && Models.Character.getVehicle() != VehicleType.DISPLAY)
+                            || !hideWhenNoGui())
+                    && isVisible();
+        return enabledTemplateCache.getValue();
```

Comment:
Please add {} after if. Braceless if statements are only allowed for returns (or break/continue) and only if it fits on a single line.

I also find this line very hard to parse. If you split it up, I think it will become much easier to read. Something like this:

```suggestion
        if (enabledTemplateCache != null && !enabledTemplateCache.hasError()) {
            return enabledTemplateCache.getValue();
        }
        
        if (!isVisible()) return false;
        if (hideWhenNoGui()) {
            return Models.WorldState.onWorld() && Models.Character.getVehicle() != VehicleType.DISPLAY;
        }
        return true;
```

Reply:
I might be misreading, but separating isVisible from hideWhenNoGui might lead to cases where there is no gui, but the overlay is still visible

Reply:
Ah yeah that's my bad, I was misreading, sorry